### PR TITLE
Add first version of amtrak script

### PR DESCRIPTION
**Pull request by sverch**
This has two modes:

- Read specific trips including train numbers from a CSV to add them up
- Dump all results for a single search query

See README for usage.

### CODE REVIEW

**Comment by SYU15** [comment]
Nice!

File: amtrakomatic/do_all_searches.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+set -euo pipefail
+
+run () {
+    echo "$@"
+    "$@"
+}
+
+
+
+#run python main.py --source NewYork --destination Boston --date  08/11/2019
```
Comment (jisaacso) [inline amtrakomatic/do_all_searches.sh]: all the comments

File: amtrakomatic/extract_expected_result.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+
+NUM_ARGS_REQUIRED=0
+if [ $# -ne "${NUM_ARGS_REQUIRED}" ]; then
+    cat <<EOF
+Usage: $0
+
+    Runs pytest and vomits the json of the expected output for whatever failed.
```
Comment (jisaacso) [inline amtrakomatic/extract_expected_result.sh]: I would suggest a real test runner, e.g., tox

File: amtrakomatic/fixup.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+echo '[]' >| test_data/seattle_chicago_08_24_2019_False_0.json
```
Comment (jisaacso) [inline amtrakomatic/fixup.sh]: why?

File: amtrakomatic/format_output.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
+"""
+Library to nicely format the result output.
+"""
+def print_result(result):
```
Comment (jisaacso) [inline amtrakomatic/format_output.py]: more descriptive name?

File: amtrakomatic/fuzzy_match.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        station_map[station_info["name"]] = station_info
+    return station_map
+
+
+def station(name):
+    """
+    Given a rough station name, does a fuzzy match on all Amtrak stations to return the proper
+    station code.
+    """
+    station_map = load_stations()
```
Comment (jisaacso) [inline amtrakomatic/fuzzy_match.py]: calls load for every search?

**Review by jisaacso — CHANGES_REQUESTED** [review]
Only you would go so far _off the rails_ . Overall looks good.

**Comment by sverch** [comment]
I've trained my whole life for this moment @jisaacso...

**Pull request merged**